### ISSUE TITLE
Create crons for email, SMS, and snapshots

### ISSUE THREAD

**Issue by gricha2380** [gricha2380/portfolioApp2380#4]
snapshot cron: daily for all users
email: daily if enabled
SMS: daily if enabled

**Comment by gricha2380** [comment]
There are a few parts to this:

- [ ] Heroku scheduler add-on: https://devcenter.heroku.com/articles/scheduler
- [ ] a npm run script in package.json to execute the cron
- [ ] A stand alone JS file to trigger a snapshot for all users in the collection. 

The standalone file is the trickiest bit. I'm not sure how to set the logic to view all documents in the collection. The current implementation gives access to the User model after authentication. The cron will have to bypass that in some currently unknown approach.

**Comment by gricha2380** [comment]
What I learned tonight at WeWork is I could create a separate route for this task and grabbing all values at once can iterate through array of all users and an inner array of all assets, firing the snapshot function for each portfolio.

**Comment by gricha2380** [comment]
Email route now computes the snapshot values; [route](https://portfolioapp2380.herokuapp.com/cron/historical). SMS and email will need similar routes.

I'm still trying to figure out how to save it though. Suggestions [here](https://stackoverflow.com/questions/38742475/what-is-the-right-approach-to-update-many-records-in-mongodb-using-mongoose) suggest a Mongo method called `bulkWrite()`, but I may be over thinking it. Maybe there's a way to do it with `findOneAndUpdate`, except I'd need to identify the value to search for.

**Comment by gricha2380** [comment]
Everything is now working as intended. A big thank you goes out to @ziggity and Kirshna for giving me a concept and the confidence to tackle this. At a specified time the system will automatically calculate and deliver portfolio updates for every user who meets the specified conditions.

**Route Details**  
Each task has it's own cron specific route and route trigger. The route is the URL, and the trigger accesses it. I added conditional logic into the routes using [findById()](http://mongoosejs.com/docs/api.html#find_find) which checks each user's delivery setting before proceeding. Particularly useful was the ability to specify and filter multiple paramaters in one argument. e.g.: `User.findById(id, 'deliverySetting emailAdddress')`. Within each cron route the same type of calculations are performed as with the normal UI route and results are passed to the same functions. SMS and Email deliver without saving, and snapshot uses `User.findOneAndUpdate` to append to user's existing historic values array.

**Scheduler Details** 
Found in [resources tab](https://dashboard.heroku.com/apps/portfolioapp2380/resources) of heroku dashboard, [the scheduler](https://devcenter.heroku.com/articles/scheduler) is a free plugin that lets me execute scripts and schedule times in UCT. Using node, I've set a trigger for each relevant task. e.g.: `node cron/SMS.js`
